index.bs

```diff
@@ -4445,6 +4465,139 @@ Of the extended attributes defined in this specification, only the [{{Exposed}}]
 </div>
 
 
+<h3 id="idl-modules">Modules</h3>
+
+A <dfn id="dfn-module" export>module</dfn> is a definition (matching
+<emu-nt><a href="#prod-Module">Module</a></emu-nt>) of a collection of
+interfaces, readonly attributes, and operations, which may be exposed
+as a logical unit.
+
+The <dfn>module specifier</dfn> of a [=module=] is the
+<a href="#prod-string">string</a> immediately following <emu-t>module</emu-t>.
+
+A module is a specification of a set of
+<dfn id="dfn-module-member" export lt="module member">module members</dfn>
+(matching <emu-nt><a href="#prod-ModuleMembers">ModuleMembers</a></emu-nt>),
+which are the [=interfaces=], [=regular operations=] and [=read only=] [=regular
+attributes=] that appear between the braces in the module declaration. These
+operations and attributes describe the behaviors packaged into the module.
+
+The IDL for modules can be split into multiple parts by using <dfn
+id="dfn-partial-module" export>partial module</dfn> definitions (matching
+<emu-t>partial</emu-t> <emu-nt><a href="#prod-Module">Module</a></emu-nt>).
+The module specifier of a partial module definition must be the same as the
+identifier of a module definition. All of the members that appear on each of
+the partial module definitions are considered to be members of the module
+itself.
+
+<pre class="syntax">
+    module <mark>"std:example"</mark> {
+      /* module_members... */
+    };
+
+    partial module <mark>"std:example"</mark> {
+      /* module_members... */
+    };
+</pre>
+
+Note: As with partial interface definitions, partial module definitions are
+intended for use as a specification editorial aide, allowing the definition
+of a module to be separated over more than one section of the document, and
+sometimes multiple documents.
+
+The order that members appear in has significance for property enumeration in
+the <a href="#es-modules">ECMAScript binding</a>.
```

Comment:
Shall we try to define a deterministic order that also works for partial to avoid introducing another place where order might differ between implementations?

Reply:
I'd be a little uncomfortable with the order being "alphabetization", as we previously discussed--this would be a little unidiomatic to require of JavaScript implementations of built-in modules.

I thought we'd handle this non-determinism in a separate, orthogonal PR. Another option would be to delay partial modules until later. I don't have any use cases in mind for partial modules, but I'm not sure if implementations are eager to split up IDL for a single module into multiple files.

(When I commented that I'm not sure if this handles partials correctly, I'm especially worried about partial implementations as module members, and if it's OK to just pretend that the partials don't exist in the later algorithms as they're already merged in.)

Reply:
I'm sorry, I don't remember that discussion, but I also don't care what method we use to determine order. What would the approach be for the separate PR?

Reply:
Perhaps https://github.com/heycam/webidl/issues/473?

Reply:
I was thinking of @tobie 's suggestion of lexicographic ordering in https://github.com/heycam/webidl/issues/432#issuecomment-334233518 . Looks like I mis-remembered it and he was actually suggesting sorting specifications by their "short name". However, @bzbarsky pointed out [issues](https://github.com/heycam/webidl/issues/432#issuecomment-334234875) with what "short name" means exactly.

Anyway, sorting *specifications* in some way doesn't seem so unidiomatic; what would be surprising is if all module members were sorted lexicographically, and the order within the specification weren't respected. But I guess no one is suggesting that.

Reply:
Ah yeah, we should preserve order within a block, that's also needed for things outside modules. So basically, we need some way to compute a sort key for each block.